File: Calculadora_basica.cs
using System;

class Program
{
    static int Sumar(int a, int b) => a + b;
    static int Restar(int a, int b) => a - b;
    static int Multiplicar(int a, int b) => a * b;
    static int Dividir(int a, int b) => b != 0 ? a / b : throw new DivideByZeroException("No se puede dividir por cero.");

    static void Menu()
    {
        while (true)
        {
            Console.WriteLine("\nSeleccione una operación:");
            Console.WriteLine("1. Sumar");
            Console.WriteLine("2. Restar");
            Console.WriteLine("3. Multiplicar");
            Console.WriteLine("4. Dividir");
            Console.WriteLine("5. Salir");
            Console.Write("Opción: ");

            string opcion = Console.ReadLine();
            if (opcion == "5") break;

            Console.Write("Ingrese el primer número: ");
            int num1 = int.Parse(Console.ReadLine());

            Console.Write("Ingrese el segundo número: ");
            int num2 = int.Parse(Console.ReadLine());

            try
            {
                int resultado = opcion switch
                {
                    "1" => Sumar(num1, num2),
                    "2" => Restar(num1, num2),
                    "3" => Multiplicar(num1, num2),
                    "4" => Dividir(num1, num2),
                    _ => throw new Exception("Opción no válida.")
                };

                Console.WriteLine($"Resultado: {resultado}");
            }
            catch (Exception e)
            {
                Console.WriteLine($"Error: {e.Message}");
            }
        }
    }

    static void Main()
    {
        Menu();
    }
}

File: Calculadora_basica.py
def suma(a, b):
    return a + b

def resta(a, b):
    return a - b

def multiplicacion(a, b):
    return a * b

def division(a, b):
    if b == 0:
        return "Error: Division por cero"
    else:
        return a / b

def menu():
    print("1. Suma")
    print("2. Resta")
    print("3. Multiplicacion")
    print("4. Division")
    opcion = int(input("Ingrese la opcion deseada: "))
    return opcion

opcion = menu()

if opcion == 1:
    digito1 = int(input("Ingrese un numero para calcular: "))
    digito2 = int(input("Ingrese un numero para calcular: "))
    resultado = suma(digito1, digito2)
    print(f"La suma de {digito1} y {digito2} es: {resultado}")

elif opcion == 2:
    digito1 = int(input("Ingrese un numero para calcular: "))
    digito2 = int(input("Ingrese un numero para calcular: "))
    resultado = resta(digito1, digito2)
    print(f"La resta de {digito1} y {digito2} es: {resultado}")

elif opcion == 3:
    digito1 = int(input("Ingrese un numero para calcular: "))
    digito2 = int(input("Ingrese un numero para calcular: "))
    resultado = multiplicacion(digito1, digito2)
    print(f"La multiplicacion de {digito1} y {digito2} es: {resultado}")

elif opcion == 4:
    digito1 = int(input("Ingrese un numero para calcular: "))
    digito2 = int(input("Ingrese un numero para calcular: "))
    resultado = division(digito1, digito2)
    print(f"La division de {digito1} y {digito2} es: {resultado}")